### PR TITLE
chore: add 1.9-stable as test reference

### PR DESCRIPTION
**Pull request by DnPlas**
Add CKF 1.9 stable as test reference so SQA can perform testing on all the combinations from the "ckf-1.9-stable" object.

### CODE REVIEW

File: test-reference/versions.json [hunk 1]
```diff
@@ -1,4 +1,30 @@
 {
+  "ckf-1.9-stable": {
+    "ckf":"1.9/stable",
+    "k8s": {
+      "charmed-kubernetes": {
+        "min-supported": "1.24",
```
Comment (orfeas-k) [inline test-reference/versions.json:6]: Min supported is 1.27, isn't it?

Comment (orfeas-k) [reply test-reference/versions.json:6]: same for all k8s substrates

File: test-reference/versions.json [hunk 2]
```diff
@@ -1,4 +1,30 @@
...
+      },
+      "eks": {
+        "min-supported": "1.24",
+        "max-supported": "1.30"
+      }
+    },
+    "juju": {
+      "current-edge": "3.4/edge",
+      "current-stable": "3.4/stable",
+      "future-beta": "3.6/beta"
```
Comment (orfeas-k) [inline test-reference/versions.json:25]: future-edge? Couldn't we use 3.5/edge for now?

**Review by orfeas-k — COMMENTED** [review]
Left some comments. Not sure if it's in scope but I think we should make the same updates (k8s version + maybe juju versions) for latest/edge. Maybe juju versions for 1.8 as well?

File: test-reference/versions.json (hunk 1 above)
Comment (DnPlas) [reply test-reference/versions.json:6]: Good catch! Changed.

File: test-reference/versions.json (hunk 2 above)
Comment (DnPlas) [reply test-reference/versions.json:25]: The future for us will be 3.6, we plan to support that version w/o passing through 3.5.
For 3.6 the only available channel is beta.

**Comment by DnPlas** [comment]
> Left some comments. Not sure if it's in scope but I think we should make the same updates (k8s version + maybe juju versions) for latest/edge. Maybe juju versions for 1.8 as well?

The scope of this task is to add a ckf-1.9 reference that SQA can take right away, we won't tell them to test latest/edge or 1.8/edge|stable, for now the focus is entirely on ckf-1.9.

**Review by orfeas-k — APPROVED** [review]

**Pull request merged**